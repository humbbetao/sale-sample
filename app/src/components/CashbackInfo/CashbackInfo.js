import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import { Container } from '@material-ui/core'
import Typography from '@material-ui/core/Typography'
import { useSelector } from 'react-redux'
import { ReactComponent as Cofrinho } from '../../assets/cofrinho.svg'
import styled from 'styled-components'
import formattedCurrency from '../../helpers/formattedCurrency'

const CofrinhoIcon = styled(Cofrinho)`
  fill: ${(props) => props.theme.colors.red};
  width: 120px;
  max-width: 150px;
`

const useStyles = makeStyles((theme) => ({
  container: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    paddingTop: '90px',
  },

  paper: {
    boxSizing: 'border-box',
    padding: '16px',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    '& > *': {
      padding: theme.spacing(2),
    },

    [theme.breakpoints.up('md')]: {
      flexDirection: 'row',
      padding: '16px',
      justifyContent: 'center',
      alignItems: 'center',
    },
  },
}))

export default function CashbackInfo() {
  const classes = useStyles()
  const cashback = useSelector((state) => state.purchase.cashback)

  return (
    <Container classes={{ root: classes.container }} maxWidth="lg">
      <Paper classes={{ root: classes.paper }}>
        <div>
          <Typography color="textPrimary" align="left">
            Cashback
          </Typography>
          <CofrinhoIcon></CofrinhoIcon>
        </div>

        <Typography color="primary" align="left" variant="h4">
          {formattedCurrency(cashback)}
        </Typography>
      </Paper>
    </Container>
  )
}
