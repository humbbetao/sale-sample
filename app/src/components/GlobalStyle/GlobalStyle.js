import { createGlobalStyle } from 'styled-components'
import { createMuiTheme } from '@material-ui/core/styles'
import breakpoints from '../../config/breakpoints'
import colors from '../Colors'
import './index.css'

export default createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: Poppins Regular;
    font-weight: 400;
    font-size: 12px;
    line-height: 28px;
    box-sizing: border-box;
    background-color: #fff;
  }
 `

export const themeStyled = { breakpoints, colors }
export const themeMaterialUi = createMuiTheme({
  palette: {
    common: { black: '#000', white: '#fff' },
    background: { paper: '#fff', default: '#fafafa' },
    primary: {
      light: 'rgba(198, 67, 133, 1)',
      main: 'rgba(199, 69, 93, 1)',
      dark: 'rgba(146, 51, 67, 1)',
      contrastText: 'rgba(230, 230, 230, 1)',
    },
    secondary: {
      light: 'rgba(115, 235, 255, 1)',
      main: 'rgba(50, 100, 110, 1)',
      dark: 'rgba(43, 86, 95, 1)',
      contrastText: 'rgba(255, 255, 255, 1)',
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    text: {
      primary: 'rgba(60, 60, 60, 1)',
      secondary: 'rgba(112, 112, 112, 1)',
      disabled: 'rgba(201, 201, 201, 1)',
      hint: 'rgba(230, 230, 230, 1)',
    },
  },
})
