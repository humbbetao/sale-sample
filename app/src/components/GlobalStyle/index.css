@font-face {
  font-family: 'Poppins Regular';
  font-style: normal;
  font-weight: 400;
  font-display: fallback;
  src: local('Poppins Regular'), local('Poppins-Regular'),
    url(../../assets/fonts/Poppins-Regular.ttf) format('truetype');
}
@font-face {
  font-family: 'Poppins Medium';
  font-style: normal;
  font-weight: 500;
  font-display: fallback;
  src: local('Poppins Medium'), local('Poppins-Medium'),
    url(../../assets/fonts/Poppins-Medium.ttf) format('truetype');
}
@font-face {
  font-family: 'Poppins Bold';
  font-style: normal;
  font-weight: 700;
  font-display: fallback;
  src: local('Poppins Bold'), local('Poppins-Bold'),
    url(../../assets/fonts/Poppins-Bold.ttf) format('truetype');
}
