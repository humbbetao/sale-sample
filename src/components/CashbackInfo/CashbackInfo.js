import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import { Container } from '@material-ui/core'
import Typography from '@material-ui/core/Typography'
import { useSelector } from 'react-redux'
import { ReactComponent as Cofrinho } from '../../assets/cofrinho.svg'
import styled from 'styled-components'
import Button from '@material-ui/core/Button'
import Add from '@material-ui/icons/Add'

const CofrinhoIcon = styled(Cofrinho)`
  fill: ${(props) => props.theme.colors.red};
  width: 120px;
  max-width: 150px;
`

const useStyles = makeStyles((theme) => ({
  container: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    paddingTop: '90px',
  },
  paperWithoutMoney: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    alignItems: 'center',
    height: '400px',
    justifyContent: 'space-evenly',
    padding: '16px',
    boxSizing: 'border-box',
    [theme.breakpoints.up('md')]: {
      display: 'flex',
      flexDirection: 'column',
      width: '100%',
      alignItems: 'center',
      height: '400px',
      justifyContent: 'space-evenly',
    },
  },
  paper: {
    boxSizing: 'border-box',
    padding: '16px',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    '& > *': {
      padding: theme.spacing(2),
    },

    [theme.breakpoints.up('md')]: {
      flexDirection: 'row',
      padding: '16px',
      justifyContent: 'center',
      alignItems: 'center',
    },
  },
}))

function CashBackIsEmpty({ handleOnOpenDialog }) {
  const classes = useStyles()
  return (
    <Container classes={{ root: classes.container }} maxWidth="lg">
      <Paper classes={{ root: classes.paperWithoutMoney }}>
        <Typography color="textPrimary" variant="h6" align="left">
          Você não tem nenhuma compra registrada
        </Typography>
        <CofrinhoIcon></CofrinhoIcon>

        <Button
          color="primary"
          component="button"
          variant="outlined"
          size="large"
          onClick={handleOnOpenDialog}
          startIcon={<Add />}
        >
          Nova Compra
        </Button>
      </Paper>
    </Container>
  )
}

export default function CashbackInfo({ isEmpty, handleOnOpenDialog }) {
  const classes = useStyles()
  const cashback = useSelector((state) => state.purchase.cashback)

  if (isEmpty) {
    return <CashBackIsEmpty handleOnOpenDialog={handleOnOpenDialog} />
  }
  return (
    <Container classes={{ root: classes.container }} maxWidth="lg">
      <Paper classes={{ root: classes.paper }}>
        <div>
          <Typography color="textPrimary" align="left">
            Cashback
          </Typography>
          <CofrinhoIcon></CofrinhoIcon>
        </div>

        <Typography
          color="primary"
          align="left"
          variant="h4"
        >{`R$ ${cashback}`}</Typography>
      </Paper>
    </Container>
  )
}
