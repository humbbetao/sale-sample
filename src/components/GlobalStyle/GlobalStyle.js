import { createGlobalStyle } from 'styled-components'
import breakpoints from '../../config/breakpoints'
import colors from '../Colors'

export default createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
 `

export const theme = { breakpoints, colors }
