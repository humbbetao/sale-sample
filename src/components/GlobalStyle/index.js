export { default, themeStyled, themeMaterialUi } from './GlobalStyle'
