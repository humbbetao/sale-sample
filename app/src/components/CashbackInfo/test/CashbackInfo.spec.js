import React from 'react'
import { mount } from 'enzyme'
import CashbackInfo from '../CashbackInfo'
import { themeStyled } from '../../GlobalStyle'
import { ThemeProvider as ThemeStyled } from 'styled-components'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
const mockStore = configureMockStore([])

const store = mockStore({ purchase: { cashback: 0, purchases: [] } })
describe('CashbackInfo', () => {
  it('should render correctly', () => {
    const purchase = {}
    let wrapper = mount(
      <Provider store={store}>
        <ThemeStyled theme={themeStyled}>
          <CashbackInfo purchase={purchase} />
        </ThemeStyled>
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toHaveLength(1)
  })
})
