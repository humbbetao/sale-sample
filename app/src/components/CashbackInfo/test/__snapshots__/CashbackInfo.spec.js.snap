// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CashbackInfo should render correctly 1`] = `
<WithStyles(ForwardRef(Container))
  classes={
    Object {
      "root": "makeStyles-container-1",
    }
  }
  maxWidth="lg"
>
  <ForwardRef(Container)
    classes={
      Object {
        "disableGutters": "MuiContainer-disableGutters",
        "fixed": "MuiContainer-fixed",
        "maxWidthLg": "MuiContainer-maxWidthLg",
        "maxWidthMd": "MuiContainer-maxWidthMd",
        "maxWidthSm": "MuiContainer-maxWidthSm",
        "maxWidthXl": "MuiContainer-maxWidthXl",
        "maxWidthXs": "MuiContainer-maxWidthXs",
        "root": "MuiContainer-root makeStyles-container-1",
      }
    }
    maxWidth="lg"
  >
    <div
      className="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
    >
      <WithStyles(ForwardRef(Paper))
        classes={
          Object {
            "root": "makeStyles-paper-2",
          }
        }
      >
        <ForwardRef(Paper)
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "outlined": "MuiPaper-outlined",
              "root": "MuiPaper-root makeStyles-paper-2",
              "rounded": "MuiPaper-rounded",
            }
          }
        >
          <div
            className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div>
              <WithStyles(ForwardRef(Typography))
                align="left"
                color="textPrimary"
              >
                <ForwardRef(Typography)
                  align="left"
                  classes={
                    Object {
                      "alignCenter": "MuiTypography-alignCenter",
                      "alignJustify": "MuiTypography-alignJustify",
                      "alignLeft": "MuiTypography-alignLeft",
                      "alignRight": "MuiTypography-alignRight",
                      "body1": "MuiTypography-body1",
                      "body2": "MuiTypography-body2",
                      "button": "MuiTypography-button",
                      "caption": "MuiTypography-caption",
                      "colorError": "MuiTypography-colorError",
                      "colorInherit": "MuiTypography-colorInherit",
                      "colorPrimary": "MuiTypography-colorPrimary",
                      "colorSecondary": "MuiTypography-colorSecondary",
                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                      "displayBlock": "MuiTypography-displayBlock",
                      "displayInline": "MuiTypography-displayInline",
                      "gutterBottom": "MuiTypography-gutterBottom",
                      "h1": "MuiTypography-h1",
                      "h2": "MuiTypography-h2",
                      "h3": "MuiTypography-h3",
                      "h4": "MuiTypography-h4",
                      "h5": "MuiTypography-h5",
                      "h6": "MuiTypography-h6",
                      "noWrap": "MuiTypography-noWrap",
                      "overline": "MuiTypography-overline",
                      "paragraph": "MuiTypography-paragraph",
                      "root": "MuiTypography-root",
                      "srOnly": "MuiTypography-srOnly",
                      "subtitle1": "MuiTypography-subtitle1",
                      "subtitle2": "MuiTypography-subtitle2",
                    }
                  }
                  color="textPrimary"
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft"
                  >
                    Cashback
                  </p>
                </ForwardRef(Typography)>
              </WithStyles(ForwardRef(Typography))>
              <Styled(Component)>
                <ForwardRef(SvgCofrinho)
                  className="sc-AxjAm bVwRyt"
                >
                  <svg
                    className="sc-AxjAm bVwRyt"
                  >
                    cofrinho.svg
                  </svg>
                </ForwardRef(SvgCofrinho)>
              </Styled(Component)>
            </div>
            <WithStyles(ForwardRef(Typography))
              align="left"
              color="primary"
              variant="h4"
            >
              <ForwardRef(Typography)
                align="left"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
                color="primary"
                variant="h4"
              >
                <h4
                  className="MuiTypography-root MuiTypography-h4 MuiTypography-colorPrimary MuiTypography-alignLeft"
                >
                  R$ 0.00
                </h4>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </div>
  </ForwardRef(Container)>
</WithStyles(ForwardRef(Container))>
`;
